demo_nodes_cpp/src/topics/listener.cpp

```diff
@@ -32,13 +34,14 @@ void print_usage()
 
 void chatterCallback(const std_msgs::msg::String::SharedPtr msg)
 {
-  std::cout << "I heard: [" << msg->data << "]" << std::endl;
+  RCLCPP_INFO(g_logger_name, "I heard: [%s]", msg->data.c_str());
```

Comment:
I'm not really a fan of this but it's the most direct approach for demos with one node. Other options here are to change the callback signature and use std::bind (it will be a bit clunky because of https://github.com/ros2/rclcpp/issues/273, so I avoided it for our "minimalist" demo) or add a standard callback signature to rclcpp that includes the logger, but if everyone is using composition then that signature "shouldn't" be needed.

Speaking of which, it's probably time to convert these demos to at least inherit from node, which will remove this less-than-ideal logging practice. I'll do that.

Reply:
I opened https://github.com/ros2/demos/pull/192 and added bd071b53f5380db71501d82d2957523bf2b89540 as a less-controversial way to pass "node loggers" to free function